### PR TITLE
Send get filters message if we haven't cached any yet

### LINKED ISSUE #1059: Filter headers doesn't continue syncing if you kill node during IBD of filters
If you stop the node during IBD of filters, and restart your node, you will not start syncing filter headers again when you restart the node. This means that your node is only partially synced with no way to restart the sync process.

### PR DESCRIPTION
**Pull request by benthecarman**
Fixes #1059

### CODE REVIEW

**Comment by Christewart** [comment]
Needs rebase

File: node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala [hunk 1]
```diff
@@ -209,7 +209,9 @@ case class DataMessageHandler(
                        "which is less than max. This means we are synced,",
                        "not requesting more.")
                     .mkString(" "))
-                if (appConfig.isNeutrinoEnabled && !syncing)
+                if (
```
Comment (Christewart) [inline node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala:216]: since this is kind of a nasty expression, can you just put a comment above the `if` saying what this checking in plain english?

Comment (Christewart) [reply node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala:216]: maybe link to the issue in source code too?

**Comment by benthecarman** [comment]
Rebased and added comment

**Review by Christewart — APPROVED** [review]
